lib/scripts.js

```diff
@@ -371,6 +371,68 @@ var scripts = {
 
     return execScript.apply(scripts, args);
   },
+
+  cleanJobsInSet: function(queue, set, ts) {
+    var command;
+    var hash;
+    switch(set) {
+      case 'completed':
+      case 'failed':
+        command = 'local JOBS = redis.call("SMEMBERS", KEYS[1])';
+        hash = 'cleanSet';
+        break;
+      case 'wait':
+      case 'active':
+      case 'paused':
+        command = 'local JOBS = redis.call("LRANGE", KEYS[1], 0, -1)';
+        hash = 'cleanList';
+        break;
+      case 'delayed':
+        command = 'local JOBS = redis.call("ZRANGE", KEYS[1], 0, -1)';
+        hash = 'cleanOSet';
+        break;
+    }
+
+    var script = [
+      command,
+      'local deleted = {}',
+      'local jobTS',
+      'for _, job in ipairs(JOBS) do',
+      '  if (redis.call("EXISTS", ARGV[1] .. job ..  ":lock") == 0) then',
+      '    jobTS = redis.call("HGET", ARGV[1] .. job, "timestamp")',
+      '    if(not jobTS or jobTS < ARGV[2]) then',
+      '      redis.call("LREM", ARGV[1] .. ARGV[3], 0, job)',
+      '      redis.call("LREM", ARGV[1] .. ARGV[4], 0, job)',
+      '      redis.call("ZREM", ARGV[1] .. ARGV[5], job)',
+      '      redis.call("LREM", ARGV[1] .. ARGV[6], 0, job)',
+      '      redis.call("SREM", ARGV[1] .. ARGV[7], job)',
+      '      redis.call("SREM", ARGV[1] .. ARGV[8], job)',
+      '      redis.call("DEL", ARGV[1] .. job)',
```

Comment:
(Im just thinking that the loop could be made 5 or 6 times faster if only removing the job one place)